blog/executor-plugins.md

```diff
@@ -0,0 +1,348 @@
+
+## Goal
+
+To make it simple but functional, I will show you how to develop an executor that creates an issue for failing Kubernetes resources such as Job, Deployment, StatefulSet, and Pod.
+
+![The final executor demo](./assets/gh-demo.gif)
+
+### Why it's worth it
+
+Because with just a few lines of code, we will automate the process of creating a GitHub issue that out-of-the box contains Kubernetes-specific information useful for further debugging. All of that, directly from Slack, Discord, Mattermost, or MS Teams! No need for connecting to your cluster in your terminal, installing and running `kubectl` commands and copy-pasting fetched details into your browser.
+
+Instead, you will be able to type `@Botkube gh create issue pod/foo` from any device that has access to your chat, including mobile apps.
+
+## Prerequisites
+
+- Access to a Kubernetes cluster
+  :::info
+  To create a local cluster for testing purposes using [`k3d`](https://k3d.io/v5.0.1/#installation), run:
+  ```bash
+  k3d cluster create
+  ```
+  :::
+- [Botkube installed and configured](https://docs.botkube.io/installation/).
+- Basic understanding of the Go language
+- [Go](https://go.dev/doc/install) at least 1.18.
+
+## What's under the hood
+
+To understand better what we will develop, I want to give you a bigger picture of the Botkube plugin system. The below animation focuses only on the executor part, but it's almost the same for sources.
+
+![Plugin system architecture animation](./assets/arch-executor-plugin.gif)
+
+The new part is a plugin repository that we [introduced in `0.17`](https://botkube.io/blog/botkube-v017-release-notes). Plugin repository is a place where you store your plugin binaries and index files. Any static file server can be used, and in this blog post we will use a GitHub release. It’s similar to what you know from the Helm ecosystem.
+
+The plugin manager consumes user's configuration, and downloads and starts **only** enabled plugins from a given repository. Plugins are running directly on the Kubernetes cluster where the Botkube core was installed.
+
+Such approach allows us to decouple the Botkube core and its extensions. Thanks to that, we can:
+
+- Avoid having the Botkube core crash if a given plugin malfunctions
+- Write extensions in any language as gRPC is used
+
+From the end-user's perspective, you can:
+
+- Specify and use multiple plugin repositories at the same time
+- Enable different plugin versions within the same Botkube version
+
+To learn more, see [Botkube architecture](https://docs.botkube.io/architecture/).
+
+## Step-By-Step Instructions
+
+To quickly onboard you to Botkube plugins, we maintain the [kubeshop/botkube-plugins-template](https://github.com/kubeshop/botkube-plugins-template) repository that has all batteries included. Our first step is to bootstrap your own GitHub repository.
+
+:::tip
+**TODO:** Embed a ~5min video where we show all the steps from this how-to blog post.
```

Comment:
I can record a dedicated video, once the blog post will be approved, and all final changes will be applied.